crisp/analysis.py

```diff
@@ -219,14 +219,26 @@ def inline_errors(
 
 # We always check out the compile_commands.json at a consistent path, in case
 # it contains relative paths.
-COMPILE_COMMANDS_PATH = 'build/compile_commands.json'
+COMPILE_COMMANDS_PATH = "compile_commands.json"
 
 @analysis
-def _cc_cmake_impl(cfg: Config, mvir: MVIR, sb: Sandbox,
-        c_code: TreeNode, cmd: list[str]) -> CompileCommandsOpNode:
+def _cc_cmake_impl(
+    cfg: Config, mvir: MVIR, sb: Sandbox, c_code: TreeNode
+) -> CompileCommandsOpNode:
+    src_dir = sb.join(cfg.relative_path(cfg.transpile.cmake_src_dir))
+    build_dir = sb.join("build")
+    setup_cmd = ["cmake", "-B", build_dir, src_dir]
+    build_cmd = ["bear", "--", "cmake", "--build", build_dir]
+    cmd = [*setup_cmd, "&&", *build_cmd]
+
     sb.checkout(c_code)
-    exit_code, logs = sb.run(cmd)
 
+    exit_code, logs = sb.run(setup_cmd)
+    if exit_code == 0:
+        new_exit_code, new_logs = sb.run(build_cmd)
+        exit_code = new_exit_code
+        logs += new_logs
```

Comment:
In other places I've been adding a `b'\n\n'` separator between old and new logs so they don't run together.

Since this is becoming a common pattern, maybe we should add a helper on `Sandbox` to run several commands in sequence.